NDTensors/ext/NDTensorsMetalExt/typeparameteraccessors.jl

```diff
@@ -0,0 +1,20 @@
+# # `TypeParameterAccessors.jl` overloads.
+using NDTensors.TypeParameterAccessors:
+  TypeParameterAccessors, Position, parameter, position, set_parameter
+## TODO this seems like a `GPUArrays` generic function
+storagemode(T::Type{<:MtlArray}) = parameter(T, 3)
+## TODO this seems like a `GPUArrays` generic function
+TypeParameterAccessors.position(::Type{<:MtlArray}, ::typeof(storagemode)) = Position(3)
+
+function TypeParameterAccessors.default_parameter(::Type{<:MtlArray}, ::typeof{eltype})
+  return Float32
+end
+TypeParameterAccessors.default_parameter(::Type{<:MtlArray}, ::typeof{ndims}) = 1
+function TypeParameterAccessors.default_parameter(::Type{<:MtlArray}, ::typeof{storagemode})
+  return Metal.DefaultStorageMode
+end
+
+## TODO this seems like a `GPUArrays` generic function
```

Comment:
I like this idea of making these `storagemode` function generic to `GPUArraysCore.AbstractGPUArray`.